core/connectionlistmodel.cpp

```diff
@@ -7,29 +7,15 @@ ConnectionListModel::ConnectionListModel(QObject *parent) :
 {
 }
 
-QHash <int, QByteArray> ConnectionListModel::roleNames() const
-{
-	QHash<int, QByteArray> roles;
-	roles[AddressRole] = "address";
-	return roles;
-}
-
 QVariant ConnectionListModel::data(const QModelIndex &index, int role) const
 {
 	if (index.row() < 0 || index.row() >= m_addresses.count())
 		return QVariant();
-	if (role != AddressRole)
+	if (role != Qt::DisplayRole)
 		return QVariant();
 	return m_addresses[index.row()];
 }
 
-QString ConnectionListModel::address(int idx) const
-{
-	if (idx < 0 || idx >> m_addresses.count())
```

Comment:
`idx >> m_addresses.count()` ??

Reply:
Yep, wasn't that brilliant?
I have no excuse.
And of course no static checker ever noticed this perfectly valid statement.